promise/src/App.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
+import {Header} from "./components/header";
+import {SearchBar} from "./components/search-bar";
+import {List} from "./components/list";
+
+const APP_ID = '6d8762c4'; 
+const APP_KEY = 'ded0f48f5bf62332e488f4731ad84dfb';
+const BASE_ENDPOINT = `https://api.edamam.com/search?app_id=${APP_ID}&app_key=${APP_KEY}&from=0&to=25`;
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state={
+      searchTerm:"cake",
+      recipes:[],
+      error:"",
+      loading:false
+    }   
+  }
+
+  componentDidMount() {this.recipeSearch();}
+
+  updateSearchTerm = e => this.setState({searchTerm:e.target.value});
+
+  recipeSearch = e => {
+    this.setState({error:""});
+    this.setState({loading:true});
```

Comment:
Keep in mind every time we run recipeSearch, we are clearing error, and loading. 

The purpose of `recipeSearch` should be a single effect, fetching. The side effects of clearing error, and setting loading, should be handled by the function that is in charge of calling the fetch, `componentDidMount`

what it should look like is maybe something like

```
componentDidMount() {
  this.clearState();
  this.recipeSearch()
    .then(this.loading(false))
    .then(this.mapRecipesToState)
    .catch(this.errorMessage);
}
```

Each of these functions is responsible for their own world 😄. We want to make sure that the function singularly responsible for one thing. What do you think?

Reply:
Thank you! :)